### PR TITLE
Changed bundler to Rollup

### LINKED ISSUE #266: Change bundler to Rollup
@amaldinesh7 _a

### PR DESCRIPTION
**Pull request by amaldinesh7**
Fixes #266 

- Changed bundler to Rollup
- Removed old Webpack config files
- Removed unused dependencies
- Moved highlight.js to peer dependency and mentioned the same in docs.

@labeebklatif _a Please review.

cc: @AbhayVAshokan @karthiknmenon

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -89,36 +80,49 @@
...
+    "tippy.js": "^6.3.1",
     "webpack": "5.64.3",
     "webpack-cli": "4.9.1",
     "webpack-dev-server": "4.7.4"
   },
   "peerDependencies": {
+    "highlight.js": "^11.3.1",
     "react": "17.0.2",
     "react-dom": "17.0.2"
   },
```
Comment (AbhayVAshokan) [inline package.json:122]: I think we can move a few more `dependencies` to `peerDependencies`.  These packages are always present in the parent project.
```suggestion
  "peerDependencies": {
    "@bigbinary/neeto-icons": "^1.8.25",
    "axios": "^0.24.0",
    "classnames": "^2.3.1",
    "highlight.js": "^11.3.1",
    "react": "17.0.2",
    "react-dom": "17.0.2"
  },
```

File: package.json [hunk 2]
```diff
@@ -89,36 +80,49 @@
...
+    "rollup": "^2.76.0",
+    "rollup-plugin-analyzer": "^4.0.0",
+    "rollup-plugin-peer-deps-external": "^2.2.4",
+    "rollup-plugin-styles": "^4.0.0",
+    "rollup-plugin-terser": "^7.0.2",
+    "rollup-plugin-visualizer": "^5.6.0",
     "sass": "^1.26.11",
     "sass-loader": "^10.0.2",
     "style-loader": "^2.0.0",
     "tailwindcss": "npm:@tailwindcss/postcss7-compat",
```
Comment (AbhayVAshokan) [inline package.json:112]: I think we can remove `tailwindcss` from the dependencies entirely. It is not used anywhere in the `lib` folder. As far as `storybook` is concerned. I believe there is [this line](https://github.com/bigbinary/neeto-editor-tiptap/blob/c9a74b04573faa02ab9822a3177822392d1cfd1d/.storybook/preview-head.html#L1-L4) loads Tailwindcss (if any tailwind classes exist).

**Review by AbhayVAshokan — COMMENTED** [review]
@amaldinesh7 I think you might have accidentally removed the `dependencies` key from `package.json`. Now all the packages come under `devDependencies`.

**Comment by AbhayVAshokan** [comment]
@amaldinesh7 a couple more changes:
1. Since `webpack-build` script no longer exists, please change [this line](https://github.com/bigbinary/neeto-editor-tiptap/blob/869b2abbc393aa79ea969b906d720a71958039f1/.github/workflows/publish.yml#L21) as well.
2. A `stats.html` file is generated on each build. It may be added to `.gitignore`.

File: package.json (hunk 1 above)
Comment (amaldinesh7) [reply package.json:122]: We've to consider neetoEditor is an open-source package. We shouldn't be making any changes with the idea of using this package only in neeto environment.

I was considering replacing axios with fetch, as there are only a few use cases.

**Comment by amaldinesh7** [comment]
> @amaldinesh7 I think you might have accidentally removed the `dependencies` key from `package.json`. Now all the packages come under `devDependencies`.

I've intentionally changed it. We don't need dependencies as we are serving the bundle.

**Comment by amaldinesh7** [comment]
@AbhayVAshokan _a Resolved comments. Please check.

**Review by AbhayVAshokan — APPROVED** [review]
@labeebklatif _a

**Comment by labeebklatif** [comment]
@amaldinesh7 A doubt. We are committing build file `dist/editor-content.min.css`. Is this intentional? required for static asset serving or something?

**Review by labeebklatif — APPROVED** [review]
@amaldinesh7 _a A doubt. We are committing build file `dist/editor-content.min.css`. Is this intentional? required for static asset serving or something? 
Other changes LGTM

**Comment by AbhayVAshokan** [comment]
Importing `dist/editor-content.min.css` from CDN is necessary for obtaining styles for the `EditorContent` component.
Ref: https://neeto-editor.onrender.com/?path=/docs/walkthroughs-output--page

I just did a `yarn build` before pushing the last commit to generate the same using Rollup.

**Pull request merged**